### DISCUSSION TITLE
Camera Focal Length and camera calibration in Sionna

### DISCUSSION THREAD

**Discussion by Leeyyhh** [NVlabs/sionna#649] [General]
Hi Everyone,
I’m currently working with the Camera class in Sionna.
I’d like to understand if Sionna’s Camera class uses traditional camera configurations to generate the images. My project requires using focal length and image resolution (instrinsic matrix)to accurately deduce object locations within the scene. But I couldn’t find explicit information on these parameters in the documentation.
Thanks a lot for your help!!

**Comment by merlinND** [comment]
Hello @Leeyyhh,

Sionna uses the rendering framework [Mitsuba](https://github.com/mitsuba-renderer/mitsuba3/) under-the-hood, and so renderings use a perspective camera that has a 3D pose (extrinsincs) as well as intrinsics.
Documentation: https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_sensors.html#perspective-pinhole-camera-perspective

That being said, I'm not sure I understand the goal.
If you can render the Sionna scene, then the location of all objects in that scene are already known, you wouldn't need to go through the camera to deduce their location?

**Reply by Leeyyhh** [reply]
Thank you for clarifying! My goal in our project is to simulate a real-life detection process based on images. Accessing locations through the camera provides a more realistic setup, including potential blockage preventing the target from being detected.  Your timely answer is a great help! :)

**Reply by merlinND** [reply]
Interesting! Then I would definitely recommend having a look at the Mitsuba documentation; if the goal is to simulate a real detection process you could improve the realism of the rendered scenes.

**Reply by Leeyyhh** [reply]
Hello, I was wondering if it might be possible to adjust the default camera's  FoV in the sionna?

**Reply by merlinND** [reply]
Hello @Leeyyhh,

Yes, have a look at the function signature of `scene.render()`:

https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/sionna/rt/scene.py#L1530-L1533

The `fov` argument controls the FoV in degrees.

**Reply by Leeyyhh** [reply]
Thanks for the timely assistance! It really solved my problem and made everything much clearer. :)

**Reply by merlinND** [reply]
Glad it helped!
